### DISCUSSION TITLE
Problem with get Nuxt Layer from private repo in self-hosted GitLab

### DISCUSSION THREAD

**Discussion by teddycoder9** [nuxt/nuxt#27486] [Questions]
I'm having trouble using Nuxt Layer when trying to connect to a private GitLab repository that is hosted on a corporate server.

I have created a new private repository containing the fresh Nuxt 3 project.
I created a new layer locally with the command `npx nuxi init --template layer nuxt-layer`

Now I want to extend in my local application to the application located in my GitLab.

Based on the documentation https://nuxt.com/docs/guide/going-further/layers I set the environment variables:

```
GIGET_GITLAB_URL=https://gitlab.mycompany.xyz
GIGET_AUTH=my_token_here
```

and the extension array:

```
extends: [
    ["git@gitlab.mycompany.xyz:user/repo_name.git", { install: true }]
],
```

Unfortunately, I get this error when running the project:

```
WARN  Cannot extend config from git@gitlab.mycompany.xyz:my_user/repo_name.git in C:/Users/x/Desktop/layer-app   
```

How to properly connect to such a repository? And is it possible to download the repository according to the selected tag (version the layer), e.g. `https://gitlab.mycompany.xyz/repo_name/-/tags/v1.2.3`?

**Comment by saschafuchs** [comment]
Try this

```
extends: [
    ["git@gitlab.mycompany.xyz:user/repo_name.git", { install: true, auth: process.env.GIGET_AUTH }]
],```

**Comment by AloisSeckar** [comment]
We are first building and publishing the base layer as a package into GitLab's internal package registry and then extending from it, not from the repo itself.

**Comment by avxkim** [comment]
It doesn’t work with self-hosted gitlab instance. You have to manually fetch repo with giget, layers documentation is poor for this part.

**Comment by molaei2000** [comment]
same issue

**Comment by NtchPlayer** [comment]
Hi !
Any news about this issue ?

It's seems `GIGET_GITLAB_URL` ENV. don't work (It didn't replace, https://gitlab.com with my value).

I make a complete report of this issue on giget repo : https://github.com/unjs/giget/issues/69

**Comment by karnauhmax** [comment]
I came across the same problem. By examining the source code of Giget, I discovered that it doesn't support GitLab subgroups. This means that it worked fine when I tried to download gitlab:group-name/repo-name, but failed when I tried to download gitlab:group-name/subgroup-name/repo-name#branch.

There is already an existing [PR](https://github.com/unjs/giget/pull/246) to fix that issue

**Reply by NtchPlayer** [reply]
I fix my problem by transforming my layer into packages: https://nuxt.com/docs/4.x/guide/going-further/layers#npm-package.
I download them directly from gitlab.

In my case, my repo are not in a subgroup, but it don't work either, so I don't know if I do something wrong... But anyway it's work for me now with package and I think it's more clean like that !
